source/fundamentals/crud/read-operations/watch.txt

```diff
@@ -0,0 +1,235 @@
+
+   db := client.Database("tea")
+
+   pipeline := bson.D{{"$match", bson.D{{"operationType", "delete"}}}}
+   changeStream, err := db.Watch(context.TODO(), mongo.Pipeline{pipeline})
+   if err != nil {
+      panic(err)
+   }
+   defer changeStream.Close(context.TODO())
+
+   for changeStream.Next(context.TODO()) {
+      fmt.Println(changeStream.Current)
+   }
+
+If you delete the ``tea.ratings`` document with a ``name`` value of 
+"``White Peony``" in a separate shell, this code will output the following:
+
+.. code-block:: go
+   :copyable: false
+   
+   {"_id": {"_data": "8262E0051B000000102B022C0100296E5A1004FB2444E03F9A4C5AB
+   3C7E868E387116846645F6964006462DFFF538BC04F3816B8BCD10004"},"operationType":
+   "delete","clusterTime": {"$timestamp":{"t":"1658848539","i":"16"}},"ns": 
+   {"db": "tea","coll": "ratings"},"documentKey": {"_id": {"$oid":"62dfff538bc04f3816b8bcd1"}}}
+
+.. note::
+
+   The ``Watch()`` method was called on the ``tea`` database, so the code outputs
+   new delete operations in any ``tea`` collection.
+
+
+Opts Parameter
+~~~~~~~~~~~~~~
+
+You can use an opts parameter to modify the behavior of the ``Watch()`` method. This
+parameter is optional and can remain ``nil``. 
+
+You can specify the following options in the opts parameter:
+
+- ``ResumeAfter``
+- ``StartAfter``
+- ``FullDocument``
+- ``BatchSize``
+- ``MaxAwaitTime``
+- ``Collation``
+- ``StartAtOperationTime``
+
+For more information on these fields, visit the 
+:manual:`MongoDB manual </reference/method/Mongo.watch/#mongodb-method-Mongo.watch>`.
+
+Example
+```````
+
+The following example calls the ``Watch()`` method on the ``tea.ratings`` collection. It 
+specifies the ``FullDocument`` opts parameter to output a copy of the entire modified document:
+
+.. code-block:: go
+
+	coll := client.Database("tea").Collection("ratings")
+	opts := options.ChangeStream().SetFullDocument(options.UpdateLookup)
+
+	changeStream, err := coll.Watch(context.TODO(), mongo.Pipeline{}, opts)
+	if err != nil {
+		panic(err)
+	}
+	defer changeStream.Close(context.TODO())
+
+	for changeStream.Next(context.TODO()) {
+		fmt.Println(changeStream.Current)
+	}
+
+If you update the ``rating`` value of the "``Masala``" tea from ``10`` to ``9``, 
+this code outputs the following: 
+
+.. code-block:: go
+   :copyable: false
+
+   {"_id": {"_data": "8262E01219000000062B022C0100296E5A1004FB2444E03F9A4C5AB3C7E
+   868E387116846645F6964006462DFFF3C5E043FA89EE1C04C0004"},"operationType": 
+   "update","clusterTime": {"$timestamp":{"t":"1658851865","i":"6"}},"fullDocument":
+   {"_id": {"$oid":"62dfff3c5e043fa89ee1c04c"},"type": "Masala","rating": {"$numberInt":"9"}},
+   "ns": {"db": "tea","coll": "ratings"},"documentKey": {"_id": {"$oid":"62dfff3c5e043fa89ee1
+   c04c"}},"updateDescription": {"updatedFields": {"rating": {"$numberInt":"9"}},"removedFields":
+   [],"truncatedArrays": []}}
+
+Without specifying the ``FullDocument`` option, the same update operation outputs the following:
```

Comment:
Suggestion: "If you don't specify the ``FullDocument`` option, the same update operation no longer outputs the ``fullDocument`` value."